pythonwhat/test_comp.py

```diff
@@ -0,0 +1,207 @@
+import ast
+from pythonwhat.State import State
+from pythonwhat.Reporter import Reporter
+from pythonwhat.Feedback import Feedback
+from pythonwhat.Test import Test, BiggerTest, EqualTest, InstanceTest
+from pythonwhat import utils
+from pythonwhat.utils import get_ord, get_num
+from pythonwhat.test_function_definition import test_args, test_body
+from pythonwhat.tasks import getTreeResultInProcess, getTreeErrorInProcess, ReprFail
+
+def test_list_comp(index=1,
+                   not_called_msg=None,
+                   comp_iter=None,
+                   iter_vars_names=False,
+                   incorrect_iter_vars_msg=None,
+                   body=None,
+                   ifs=None,
+                   insufficient_ifs_msg=None,
+                   expand_message=True):
+    """Test list comprehension.
+    """
+
+    state = State.active_state
+    rep = Reporter.active_reporter
+    rep.set_tag("fun", "test_list_comp")
+
+    state.extract_list_comps()
+    student_list_comps = state.student_list_comps
+    solution_list_comps = state.solution_list_comps
+
+    test_comp(index=index,
+              not_called_msg=not_called_msg,
+              comp_iter=comp_iter,
+              iter_vars_names=iter_vars_names,
+              incorrect_iter_vars_msg=incorrect_iter_vars_msg,
+              body=body,
+              ifs=ifs,
+              insufficient_ifs_msg=insufficient_ifs_msg,
+              expand_message=expand_message,
+              student_comp_list=student_list_comps,
+              solution_comp_list=solution_list_comps,
+              state=state,
+              rep=rep,
+              comp_type="list")
+
+def test_generator_exp(index=1,
```

Comment:
`test_generator_exp` and `test_list_comp` seem exactly the same except for `comp_type` (if the comment above about the parsers being the same was valid).